src/asm/rpn.c

```diff
@@ -477,6 +518,9 @@ void rpn_BinaryOp(enum RPNCommand op, struct Expression *expr,
 
 		expr->val = sym_GetValue(symbol1) - sym_GetValue(symbol2);
 		expr->isKnown = true;
+	} else if (op == RPN_AND && (constMaskVal = tryConstMask(src1, src2)) != -1) {
```

Comment:
My bad, `checkpatch` doesn't like `=` in `if` conditions. Your previous structure is fine for now then, and possible we can refactor both `isDiffConstant` and `tryConstMask` depending on #977.

Reply:
Eh, I'm fine with disregarding `checkpatch`. This'll turn into something nicer when we switch to C++, and `checkpatch` will be largely irrelevant then.

Reply:
For now we do still have it enabled, though, and it does catch some worthwhile lint issues. There will be more PRs before the C++ switch, and I'd rather just be able to check for all green.

Reply:
It only checks for violations *introduced* in a given PR/commit, so it will stay green for future PRs.